judge/runguard.cc

```diff
@@ -802,8 +805,8 @@ void setrestrictions()
 	}
 
 	/* Set additional environment variables. */
-	if (environment_variables != nullptr) {
-		char *token = strtok(environment_variables, ";");
+	for (const auto &env_var : environment_variables) {
+		char *token = strtok(strdup(env_var.c_str()), ";");
```

Comment:
Nitpick: `env_var` can actually contain multiple variables separated by `;`, so maybe rename to something like
```suggestion
	for (const auto &tokens : environment_variables) {
		char *token = strtok(strdup(tokens.c_str()), ";");
```

Reply:
Done.